projects/pgai/pgai/vectorizer/vectorizer.py

```diff
@@ -504,12 +556,38 @@ async def run(self) -> int:
                             ),
                         ),
                     )
+
+                if e.__cause__ is not None:
+                    raise e.__cause__  # noqa
+                raise e
+
+            except UriLoadingError as e:
```

Comment:
hmm handling this error at this level fails the rest of the rows in the batch too? what if 99 items succeeded and 1 failed? This seems not very good. why don't we handle these like we do in `_generate_embeddings`. specifically we should handle it like `ChunkEmbeddingError` and return but not raise.

Reply:
fixed, coming in a separated PR.

Reply:
done